iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        error = null;
+        value =null;
+        resultSetWithPath = null;
+    }
+
+    /**
+     * When DockerUtil.getInstance().getContainer(microserviceUuid) returns null
+     * then createImageSnapshot returns
+     */
+    @Test
+    public void createImageSnapshotWithNullContainer() {
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(null);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
+    }
+
+    /**
+     * When DockerUtil.getInstance().getContainer(microserviceUuid) returns a container,
+     * CommandShellExecutor.executeCommand returns a resultset with value and no error
+     *
+     */
+    @Test
+    public void createImageSnapshotWithCommandExecuteSuccess() {
+
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        value.add("path");
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
+    }
+
+    /**
+     * When DockerUtil.getInstance().getContainer(microserviceUuid) returns a container,
+     * CommandShellExecutor.executeCommand returns a resultset with error and no value
+     *
+     */
+    @Test
+    public void createImageSnapshotWithCommandExecuteFailure() {
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        error.add("error");
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
+    }
+
+
+    /**
+     * When DockerUtil.getInstance().getContainer(microserviceUuid) returns a container,
+     * CommandShellExecutor.executeCommand returns a resultset is Empty
+     *
+     */
+    @Test
+    public void createImageSnapshotWithCommandExecuteReturnsEmpty() {
+        error = new ArrayList<>();
+        value = new ArrayList<>();
+        resultSetWithPath = new CommandShellResultSet<>(value, error);
+
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(Optional.of(container));
+        when(CommandShellExecutor.executeCommand(any())).thenReturn(resultSetWithPath);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
```

Comment:
Also, test it does not call `Orchestrator#sendFileToController()`

Reply:
updated